ProcessLib/Process.h

```diff
@@ -258,24 +269,8 @@ class Process
 		}
 		assert(result && result->size() == _x->size());
 
-#ifdef USE_PETSC
-		std::unique_ptr<double[]> u(new double[_x->size()]);
-		_x->getGlobalVector(u.get());  // get the global solution
-
-		std::size_t const n = _mesh.getNNodes();
-		for (std::size_t i = 0; i < n; ++i)
-		{
-			MeshLib::Location const l(_mesh.getID(),
-			                          MeshLib::MeshItemType::Node, i);
-			auto const global_index = std::abs(  // 0 is the component id.
-			    _local_to_global_index_map->getGlobalIndex(l, 0));
-			(*result)[i] = u[global_index];
-		}
-#else
 		// Copy result
-		for (std::size_t i = 0; i < _x->size(); ++i)
-			(*result)[i] = (*_x)[i];
-#endif
+		_x->getValues(&(*result)[0]);
```

Comment:
Function ``_x->getValues(&(*result)[0])`` just fetches the values from the equation solutions, and it does not distinguish the solutions for components.  For the case of  component results, we have to modify this function as
1. Consider component in the following piece of code for variable name and the cache for the solutions to be output 
```
		// Get or create a property vector for results.
		boost::optional<MeshLib::PropertyVector<double>&> result;
		if (_mesh.getProperties().hasPropertyVector(property_name))
		{
			result = _mesh.getProperties().template getPropertyVector<double>(
			    property_name);
		}
		else
		{
			result =
			    _mesh.getProperties().template createNewPropertyVector<double>(
			        property_name, MeshLib::MeshItemType::Node);
			result->resize(_x->size());
		}
```
2. Fill the cache ``result`` for output with a similar piece of code to e previous version of the petsc code.
3. Possibly, modify vtu_interface.writeToFile(file_name) for multiple solution arrays.

I will check this issue in another PR.